docs/ops/mem_setup.md

```diff
@@ -0,0 +1,129 @@
+---
+title: "Task Manager Memory Configuration"
+nav-parent_id: ops
+nav-pos: 5
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+Apache Flink provides efficient workloads on top of the JVM by tightly controlling the memory usage of its various components.
+While the community strives to offer sensible defaults to all configurations, the full breadth of applications
+that users deploy on Flink means this isn't always possible. To provide the most production value to our users,
+Flink allows both high level and fine-grained tuning of memory allocation within clusters.
+
+* toc
+{:toc}
+
+See also task manager [configuration options](config.html#taskmanager).
+
+## Total memory
+
+The *total memory* in Flink consists of JVM heap, [managed memory](#managed-memory) and [network buffers](#network-buffers).
+[Managed memory](#managed-memory) can be either part of JVM heap or direct off-heap memory.
+For containerised deployments, the *total memory* can additionally include a [container cut-off](#container-cut-off).
+
+All other memory components are computed from the *total memory* before starting the Flink process.
+After the start, the [managed memory](#managed-memory) and [network memory](#network-buffers) are adjusted in certain cases
+based on available JVM memory inside the process (see [Adjustments inside Flink process](#adjustments-inside-flink-process)).
+
+<center>
+  <img src="{{ site.baseurl }}/fig/mem_model.svg" width="500px" alt="Simple memory model" usemap="#simple-mem-model">
+</center>
+<br />
+
+When the Flink JVM process is started from scripts in [standalone mode](deployment/cluster_setup.html),
+on [Yarn](deployment/yarn_setup.html) or [Mesos](deployment/mesos.html), the *total memory* is defined by
+the configuration option [taskmanager.heap.size](config.html#taskmanager-heap-size-1) (or deprecated `taskmanager.heap.mb`).
```

Comment:
It seems that we don't treat config keys consistently. On other pages we use backticks for their formatting.